pkgs/tools/compression/lzip/default.nix

```diff
@@ -4,14 +4,15 @@ stdenv.mkDerivation rec {
   name = "lzip-${version}";
   version = "1.20";
 
-  buildInputs = [ texinfo ];
+  nativeBuildInputs = [ texinfo ];
 
   src = fetchurl {
     url = "mirror://savannah/lzip/${name}.tar.gz";
     sha256 = "0319q59kb8g324wnj7xzbr7vvlx5bcs13lr34j0zb3kqlyjq2fy9";
   };
 
-  configureFlags = "CPPFLAGS=-DNDEBUG CFLAGS=-O3 CXXFLAGS=-O3" + stdenv.lib.optionalString stdenv.isCross " CXX=${stdenv.cc.targetPrefix}c++";
+  configureFlags = "CPPFLAGS=-DNDEBUG CFLAGS=-O3 CXXFLAGS=-O3"
+                 + stdenv.lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) " CXX=${stdenv.cc.targetPrefix}c++";
```

Comment:
Is explicit `CXX` actually needed here?

Reply:
Looks like it. I think it makes sense to make it unconditional though.

Reply:
If you make it unconditional, there's a risk that someone notices it native-compiles fine and removes it completely. So I would keep it conditional.

Reply:
@dezgeg but that's why we do https://github.com/NixOS/nixpkgs/issues/21471! :)

Reply:
No, that's not something to be expected from a standard native-compiling environment.